### PR TITLE
[Docs] Add Migration guide + some API updates

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
## Description of Change
Created Migration guide page in documentation for migration from 2.x to 3.0.
Updated SigmaDelta API docs.
Also added to some peripherals docs informations about C6/H2 + some minor fixes.

## Related links
Rebase of PR #8408

### CODE REVIEW

**Comment by P-R-O-C-H-Y** [comment]
@lucasssvaz @pedrominatel Migration guide is ready for review.
As this is a new (rebased) PR, you can just check RMT and UART part + links, as the other parts were reviewed in #8716.

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+* ``rmtSetRxMinThreshold``
+
+
+Changes in APIs
+***************
+
+* In all functions, input parameter ``rmt_obj_t* rmt`` has been changed to ``int pin``.
+* ``rmtInit`` return parameter changed to bool.
+* ``rmtInit`` input parameter ``bool tx_not_rx`` has been changed to ``rmt_ch_dir_t channel_direction``.
+* ``rmtInit`` new input parameter ``uint32_t frequency_Hz`` to set frequency of RMT channel as function ``rmtSetTick`` was removed.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:100]:
```suggestion
* ``rmtInit`` new input parameter ``uint32_t frequency_Hz`` to set frequency of RMT channel (as function ``rmtSetTick`` was removed).
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+* ``rmtWrite`` now sending data in blocking mode - only returns after sending all data or by timeout. For Async mode use new ``rmtWriteAsync`` function.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:101]:
```suggestion
* ``rmtWrite`` now sending data in blocking mode. It only returns after sending all data or through a timeout. For Async mode use the new ``rmtWriteAsync`` function.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+* ``rmtWrite`` new input parameter ``uint32_t timeout_ms``.
+* ``rmtLoop`` renamed to ``rmtWriteLooping``.
+* ``rmtRead`` input parameters changed to ``int pin, rmt_data_t* data, size_t *num_rmt_symbols, uint32_t timeout_ms``.
+* ``rmtReadAsync`` input parameters changed to ``int pin, rmt_data_t* data, size_t *num_rmt_symbols``.
+* ``rmtSetRxThreshold`` renamed to ``rmtSetRxMaxThreshold`` and input parameter ``uint32_t value`` has been changed to ``uint16_t idle_thres_ticks``.
+* ``rmtSetCarrier`` input parameters ``uint32_t low, uint32_t high`` has been changed to ``uint32_t frequency_Hz, float duty_percent``.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:107]:
```suggestion
* ``rmtSetCarrier`` input parameters ``uint32_t low, uint32_t high`` have been changed to ``uint32_t frequency_Hz, float duty_percent``.
```

**Review by lucasssvaz — APPROVED** [review]
LGTM, just some minor improvements

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  to achieve the selected frequency.
+* ``timerAttachInterrupt`` has now only 2 parameters. The ``edge`` parameter has been removed.
+
+UART
+----
+
+Removed APIs
+************
+
+* ``uartDetachPins``
```
Comment (SuGlider) [inline docs/source/migration_guide/2.x_to_3.0.rst:179]: This function is internal and not available in HardwareSerial API.
This is internal to esp32-hal-uart.h only, but user don't use such APIs.

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+  
+New APIs
+********
+
+* ``uart_get_RxPin`` used to get UART RX pin number.
+* ``uart_get_TxPin`` used to get UART TX pin number.
+
```
Comment (SuGlider) [inline docs/source/migration_guide/2.x_to_3.0.rst:186]: These are internal helper functions, not exposed in HardwareSerial class.

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+Changes in APIs
+***************
+
+* ``uartEnd`` input parameter ``uart_t* uart`` has been changed to ``uint8_t uart_num``.
+* ``uartSetMode`` input parameter ``uint8_t mode`` has been changed to ``uart_mode_t mode``.
+* ``uartSetHwFlowCtrlMode`` input parameter ``uint8_t mode`` has been changed to ``uart_hw_flowcontrol_t mode``.
```
Comment (SuGlider) [inline docs/source/migration_guide/2.x_to_3.0.rst:192]: These functions are from esp32-hal-uart.c, which are internal only. 
Arduino has HardwareSerial class used by `Serial` which is the UART interface for users.

File: docs/source/api/sigmadelta.rst [hunk 7]
```diff
@@ -14,67 +14,61 @@ ESP32 SoC Number of SigmaDelta channels
...
+This function returns ``true`` if setting the duty was successful.
+If ``false`` is returned, error occurs and duty was not set.
 
-sigmaDeltaDetachPin
-*******************
+sigmaDeltaDetach
+****************
 
-This function is used to detach pin from SigmaDelta.
+This function is used to detach a pin from SigmaDelta and deinit the channel that was attached to the pin.
```
Comment (pedrominatel) [inline docs/source/api/sigmadelta.rst:61]: deinitialize

**Review by pedrominatel — APPROVED** [review]

File: docs/source/index.rst [hunk 8]
```diff
@@ -16,6 +16,7 @@ Here you will find all the relevant information about the project.
    Guides <guides/guides>
    Tutorials <tutorials/tutorials>
    Advanced Utilities <advanced_utils>
+   Migration Guide <migration_guide/migration_guide>
```
Comment (VojtechBartoska) [inline docs/source/index.rst:19]: Migration Guide**s**

We can call this tree section more generic so we can use it later on.

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 9]
```diff
@@ -0,0 +1,192 @@
+#########################
+Migration from 2.x to 3.0
+#########################
+
+Introduction
+------------
+
+This is a guide to highlight breaking changes in the API and to help the migration of projects from versions 2.X to version 3.0 of the Arduino ESP32 core.
```
Comment (VojtechBartoska) [inline docs/source/migration_guide/2.x_to_3.0.rst:8]: add space line between line 8 and 9.

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 10]
```diff
@@ -0,0 +1,192 @@
+#########################
+Migration from 2.x to 3.0
+#########################
+
+Introduction
+------------
+
+This is a guide to highlight breaking changes in the API and to help the migration of projects from versions 2.X to version 3.0 of the Arduino ESP32 core. 
+For more information about the changes and new features, check `RELEASE NOTES <https://github.com/espressif/arduino-esp32/releases>`_.
```
Comment (VojtechBartoska) [inline docs/source/migration_guide/2.x_to_3.0.rst:9]: For more information about all changes and new features, check project `RELEASE NOTES <https://github.com/espressif/arduino-esp32/releases>`_.

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+
```
Comment (VojtechBartoska) [inline docs/source/migration_guide/2.x_to_3.0.rst:10]: What about adding heading here, something like:

**Breaking Changes in APIs**

File: docs/source/migration_guide/2.x_to_3.0.rst (hunk 9 above)
Comment (VojtechBartoska) [inline docs/source/migration_guide/2.x_to_3.0.rst:8]: Can we mentioned here that v2.X is based on ESP-IDF v4.4 and 3.0 is based on ESP-IDF 5.X?

**Review by VojtechBartoska — COMMENTED** [review]
LGTM, just minor changes.

**Comment by VojtechBartoska** [comment]
**Missing:**
- latest BLE changes
- add sentence that the examples have been updated and User can use them

**Comment by P-R-O-C-H-Y** [comment]
> **Missing:**
> 
> * latest BLE changes
> * add sentence that the examples have been updated and User can use them

@VojtechBartoska Done

**Comment by TD-er** [comment]
Still missing info about how to get the IP of a remote client from the webserver.
See: https://github.com/espressif/arduino-esp32/issues/8774

**Comment by P-R-O-C-H-Y** [comment]
> Still missing info about how to get the IP of a remote client from the webserver.
> See: #8774

There was no change in API about `remoteIP()`. So nothing to be added. Probably its not working correctly in 3.0.0 so we need to take a look :)

**Pull request merged**